spec.md

```diff
@@ -100,6 +110,18 @@ interface ExpressionStatement <: Statement {
 
 An expression statement, i.e., a statement consisting of a single expression.
 
+## Directive
+
+```js
+interface Directive <: ExpressionStatement {
+    directive: string;
+}
+```
+
+A directive from the directive prologue of a script or function.
+The `directive` property is the raw string source of the directive.
+This source includes single or double wrapping quotes and may not contain an EscapeSequence or LineContinuation..
```

Comment:
`value` should be never trusted as directive source due to escaping, so if it's not present, tool can just `return false` immediately. If node generator (whether parser or custom builder) wants to generate directive, it will be just obliged to put `raw` into it (same as, for example, with generating Literal nodes for asm.js which distinguishes `2.0` from `2`).

This way, first example will be simplified to:

```javascript
return statement.directive && (statement.expression.raw || '').slice(1, -1) === "use strict";
```

Which is not complicated and in any case you just write it once and then reuse.

P.S. `raw` is already present in nodes generated by all the existing parsers, so it's not breaking backwards compatibility but standardizing it (as we did with most "non-standard" features when ESTree was initiated).